### ISSUE TITLE
[BUG]queueName or topicName does not make any difference for ServiceBusSenderAsyncClient 

### ISSUE THREAD

**Issue by jialigit** [Azure/azure-sdk-for-java#28125]
**Describe the bug**
queueName or topicName does not make any difference for ServiceBusSenderAsyncClient . It does not make sense.

***Exception or Stack Trace***
Add the exception log and stack trace if available

**To Reproduce**
As per the following snippet: from :https://docs.microsoft.com/en-us/java/api/com.azure.messaging.servicebus.servicebussenderasyncclient?view=azure-java-stable

change the method of queueName to topicName, makes no difference.
```
// The required parameters is connectionString, a way to authenticate with Service Bus using credentials.
 // The connectionString/queueName must be set by the application. The 'connectionString' format is shown below.
 // "Endpoint={fully-qualified-namespace};SharedAccessKeyName={policy-name};SharedAccessKey={key}"
 ServiceBusSenderAsyncClient sender = new ServiceBusClientBuilder()
     .connectionString(connectionString)
     .sender()
     .queueName(queueName)
     .buildAsyncClient();

 // Creating a batch without options set, will allow for automatic routing of events to any partition.
 sender.createMessageBatch().flatMap(batch -> {
     batch.tryAddMessage(new ServiceBusMessage(BinaryData.fromBytes("test-1".getBytes(UTF_8))));
     batch.tryAddMessage(new ServiceBusMessage(BinaryData.fromBytes("test-2".getBytes(UTF_8))));
     return sender.sendMessages(batch);
 }).subscribe(unused -> {
 },
     error -> System.err.println("Error occurred while sending batch:" + error),
     () -> System.out.println("Send complete."));
```
***Code Snippet***
Add the code snippet that causes the issue.

**Expected behavior**

**Screenshots**

**Setup (please complete the following information):**
 - IDE: [e.g. IntelliJ]
 - Library/Libraries: [e.g. com.azure:azure-core:1.16.0 (groupId:artifactId:version)]
 - Java version: [e.g. 8] 
 - App Server/Environment: [e.g. Tomcat, WildFly, Azure Function, Apache Spark, Databricks, IDE plugin or anything special]
 - Frameworks: [e.g. Spring Boot, Micronaut, Quarkus, etc] 

If you suspect a dependency version mismatch (e.g. you see `NoClassDefFoundError`, `NoSuchMethodError` or similar), please check out [Troubleshoot dependency version conflict article](https://aka.ms/azsdk/java/dependency/troubleshoot) first. If it doesn't provide solution for the problem, please provide:
- verbose dependency tree (`mvn dependency:tree -Dverbose`)
- exception message, full stack trace, and any available logs

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [ ] Bug Description Added
- [ ] Repro Steps Added
- [ ] Setup information Added

**Comment by joshfree** [comment]
@ki1729 could you please follow up with @jialigit

/cc @Azure/azsdk-sb-java

**Comment by ki1729** [comment]
@jialigit This is correct, as far as the sender is concerned, they are sending ServiceBusMessages to an entity, the method used depends upon the entity the user wants to use.  AFAIK, the distinction is mostly used on the service side for message processing. As a sender client, there is no way to determine is the given entity is a queue or a topic and using a generic entityName field would introduce more confusion.

**Comment by jialigit** [comment]
> @jialigit This is correct, as far as the sender is concerned, they are sending ServiceBusMessages to an entity, the method used depends upon the entity the user wants to use. AFAIK, the distinction is mostly used on the service side for message processing. As a sender client, there is no way to determine is the given entity is a queue or a topic and using a generic entityName field would introduce more confusion.

Yes, I understand this point. This a refactor suggestion from customer. The usage here introduces some confusion if they donot know how the server side handle it.

**Comment by ki1729** [comment]
@jialigit Sorry if I missed this, but could please share some more details on the refactor suggestion? Is it to use a unified method?
And as for the confusion, shouldn't users directly use queueName(), when using a queue and topicName() when using a topic? From client's prespective, ideally there should be no need to know the implementation details of how serve side handles the sent message. Also, even if by mistake, they interchange and use the wrong method, the outcome will be the same (No error will be thrown, and the messages will be sent to the correct entity).

**Comment by jialigit** [comment]
@ki1729 Thanks. Yes, I also think clients does not need to know the details about how the server side handles the message. This is not big issue really. Unified method brings confusion either. I will sync this to the user.

**Comment by taisuke-fujimoto** [comment]
@ki1729 HI, I created the original question in https://github.com/Azure/azure-sdk-for-java/issues/26923#issuecomment-1088188876
I saw the code of ServiceBusSenderAsyncClient and noticed that entityType is used in `scheduleMessage()`, which means that entityType is needed.

If you create `scheduleMessage()` for ServiceBusTemplate in the future, I think defaultEntityType of ServiceBusTemplate will be required.

> isn't it necessary to separate the ServiceBusTemplate instance by queue and topic?

If I answer my question, now: no, future: may change to yes.

---
I don't really know what this SDK is doing, I just guessed from the code

**Comment by ki1729** [comment]
Hi @taisuke-fujimoto! Appologies for the late reply.

> If I answer my question, now: no, future: may change to yes.

You are right, currently it does not influence behavior of the sender client, but it may in the future. So setting entityType will be be needed

**Comment by taisuke-fujimoto** [comment]
@ki1729 Thanks for your reply.
There seems to be no need to change ServiceBusSenderAsyncClient.

However, ServiceBusTemplate still has some confusion.
Should I post here or back to #26923 ? @jialigit

**Comment by jialigit** [comment]
@taisuke-fujimoto Yes. Let's discuss the issues related to "ServiceBusTemplate" in #26923. Let's reopen it.

**Comment by taisuke-fujimoto** [comment]
@jialigit I got it.
